samples/boards/nrf52/sensor_temperature/src/main.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <gpio.h>
+#include <sensor.h>
+
+#define LED_PORT	DT_GPIO_LEDS_LED0_GPIO_CONTROLLER
+#define LED0		DT_GPIO_LEDS_LED0_GPIO_PIN
+#define LED1		DT_GPIO_LEDS_LED1_GPIO_PIN
+#define LED2		DT_GPIO_LEDS_LED2_GPIO_PIN
+#define LED3		DT_GPIO_LEDS_LED3_GPIO_PIN
+int led_write_values[4] = {0};
+
+/* 1000 msec = 1 sec */
+#define SLEEP_TIME	1000
+
+#define TEMPERATURE_READ_FAILED  1
+
+int get_new_temperature_reading(
+	struct device *temperature_dev,
+	struct sensor_value *val_ptr)
+{
+	if (sensor_sample_fetch_chan(temperature_dev, SENSOR_CHAN_DIE_TEMP)) {
+		printk("sensor fetch failed.\n");
+		return TEMPERATURE_READ_FAILED;
+	}
+
+	if (sensor_channel_get(temperature_dev,
+			       SENSOR_CHAN_DIE_TEMP,
+			       val_ptr)) {
+		printk("sensor channel get failed.\n");
+		return TEMPERATURE_READ_FAILED;
+	}
+
+	return 0;
+}
+
+void main(void)
+{
+	struct device *dev;
+	struct device *nrf_temperature_dev;
+	struct sensor_value temp_sensor_val;
+
+	printk("nRF52-DK Temperature Sensor Demo.\n");
+
+	/* Get LED Driver binding and set LED pins as output */
+	dev = device_get_binding(LED_PORT);
+	gpio_pin_configure(dev, LED0, GPIO_DIR_OUT);
+	gpio_pin_configure(dev, LED1, GPIO_DIR_OUT);
+	gpio_pin_configure(dev, LED2, GPIO_DIR_OUT);
+	gpio_pin_configure(dev, LED3, GPIO_DIR_OUT);
```

Comment:
So, should i be getting/creating LED Driver binding for each and every LED? 
Will this LED driver binding be "attached" to the same driver - since they are in fact connected to the same GPIO port? How do you handle this if you have more, like 32, GPIOs with LEDs? Will every single one get its one driver binding?

Reply:
Using DT_GPIO_P0_DEV_NAME from ...nordic_nrf\nrf52\dts_fixup.h. Not sure if this makes a difference as I am still assuming (/know) that all GPIOs are on the same GPIO P0 port.

Reply:
No, it does not make much difference. My point is that if you decide to get information about LEDs from DTS, it does not look good when you then assume that all the LEDs are on the same port. You should rather use `DT_GPIO_LEDS_LEDx_GPIO_CONTROLLER` for each LED. The same applies to their active state. It should also be taken from DTS, and currently the code assumes that `0` is always the active state.
On the other hand, using all this information from DTS will blow the code of this fairly simple application. So I'm not convinced it is worth it. Maybe this controlling of the LEDS could be dropped and just printing the temperature would be enough. It does not seem to be easy to see the temperature crossing the 5 or 10 degrees threshold on your desk, so maybe this LED signaling will not be that useful.

Reply:
Example removed.